include/aos/iam/modules/pkcs11.hpp

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_IAM_MODULES_PKCS11_HPP_
+#define AOS_IAM_MODULES_PKCS11_HPP_
+
+#include "hsm.hpp"
+
+#include "aos/iam/modules/config.hpp"
+
+#include "aos/common/pkcs11/pkcs11.hpp"
+#include "aos/common/tools/optional.hpp"
+#include "aos/common/uuid.hpp"
+
+namespace aos {
+namespace iam {
+namespace certhandler {
+
+/**
+ * PKCS11 configuration
+ */
+struct ModuleConfig {
+    /**
+     * Library name.
+     */
+    StaticString<cFilePathLen> mLibrary;
+
+    /**
+     * Slot ID.
+     */
+    Optional<uint32_t> mSlotID;
+
+    /**
+     * Slot index.
+     */
+    Optional<int> mSlotIndex;
+
+    /**
+     * Token label.
+     */
+    StaticString<pkcs11::cLabelLen> mTokenLabel;
+
+    /**
+     * Path to User PIN.
+     */
+    StaticString<cFilePathLen> mUserPINPath;
+
+    /**
+     * USER identifier.
+     */
+    uint32_t mUID;
+
+    /**
+     * GROUP identifier.
+     */
+    uint32_t mGID;
+
+    /**
+     * Module
+     */
+    bool mModulePathInURL;
+};
+
+/**
+ * A helper object for search operations containing the most valuable data for certhandler.
+ */
+struct SearchObject;
```

Comment:
Looks like it is private for `PKCS11Module`. Move its definition under private section of `PKCS11Module`.